services/bls_aggregation/blsagg.go

```diff
@@ -350,8 +344,28 @@ func (a *BlsAggregatorService) singleTaskAggregatorGoroutineFunc(
 				// happens..
 				continue
 			}
-			// after verifying signature we aggregate its sig and pubkey, and update the signed stake amount
+
+			// check if the operator has already signed for this digest
 			digestAggregatedOperators, ok := aggregatedOperatorsDict[taskResponseDigest]
+			if ok {
```

Comment:
there's already an `if ok {} else {}` conditional below. Probably just want to put everything in here in the else clause below?

Reply:
Wouldn't this mess with the error handling? Between this `if ok` and the other `if !ok {} else {}` we unconditionally send the result to the error channel. We could move the `if/else` here, but that would increase the amount of computation before sending a result.

Reply:
I guess the performance impact would be minimal - I'll switch to the more readable approach

Reply:
7627a5eec95d20ce0c9538fedfde374ab990696d

Reply:
scratch that - there's not really a way to move it to the  if/else without restructuring the code a decent amount because of how the error channels are consumed

Reply:
Ah my bad... ugly formatting on github made it look like all the logic was part of the same if ok {} conditional. Sorry for making you waste time here @afkbyte :\